Tests/SentryTests/Networking/SentryReachabilitySwiftTests.swift

```diff
@@ -1,7 +1,120 @@
+        XCTAssertEqual("none", SentryReachabilityTestHelper.stringForSentryConnectivity(.none))
+        XCTAssertEqual("wifi", SentryReachabilityTestHelper.stringForSentryConnectivity(.wiFi))
+        #if canImport(UIKit)
+        XCTAssertEqual("cellular", SentryReachabilityTestHelper.stringForSentryConnectivity(.cellular))
+        #endif
+    }
+    
+    func testMultipleReachabilityObservers() {
+        print("[Sentry] [TEST] creating observer A")
+        let observerA = TestSentryReachabilityObserver()
+        print("[Sentry] [TEST] adding observer A as reachability observer")
+        reachability.add(observerA)
+        
+        print("[Sentry] [TEST] throwaway reachability callback, setting to reachable")
+        reachability.triggerConnectivityCallback(.wiFi) // ignored, as it's the first callback
+        print("[Sentry] [TEST] reachability callback set to unreachable")
+        reachability.triggerConnectivityCallback(.none)
+        
+        print("[Sentry] [TEST] creating observer B")
+        let observerB = TestSentryReachabilityObserver()
+        print("[Sentry] [TEST] adding observer B as reachability observer")
+        reachability.add(observerB)
+        
+        print("[Sentry] [TEST] reachability callback set back to reachable")
+        reachability.triggerConnectivityCallback(.wiFi)
+        print("[Sentry] [TEST] reachability callback set back to unreachable")
+        reachability.triggerConnectivityCallback(.none)
+        
+        print("[Sentry] [TEST] removing observer B as reachability observer")
+        reachability.remove(observerB)
+        
+        print("[Sentry] [TEST] reachability callback set back to reachable")
+        reachability.triggerConnectivityCallback(.wiFi)
+        
+        XCTAssertEqual(5, observerA.connectivityChangedInvocations)
+        XCTAssertEqual(2, observerB.connectivityChangedInvocations)
+        
+        print("[Sentry] [TEST] removing observer A as reachability observer")
+        reachability.remove(observerA)
+    }
+    
+    func testNoObservers() {
+        let observer = TestSentryReachabilityObserver()
+        reachability.add(observer)
+        reachability.remove(observer)
+        
+        reachability.triggerConnectivityCallback(.wiFi)
+        
+        XCTAssertEqual(0, observer.connectivityChangedInvocations)
+        
+        reachability.removeAllObservers()
+    }
+    
+    func testReportSameObserver_OnlyCalledOnce() {
+        let observer = TestSentryReachabilityObserver()
+        reachability.add(observer)
+        reachability.add(observer)
+        
+        reachability.triggerConnectivityCallback(.wiFi)
+        
+        XCTAssertEqual(1, observer.connectivityChangedInvocations)
+        
+        reachability.remove(observer)
+    }
+    
+    /**
+     * We only want to make sure running the actual registering and unregistering callbacks doesn't
+     * crash.
+     */
+    func testRegisteringActualCallbacks() {
```

Comment:
Ups forgot one comment.

`m`: I think it would be great to have one test not using the `skipRegisteringActualCallbacks` and actually running the code for the `NWPathMonitor` and validating with a test expectation or something that we at least get one update via the `pathUpdateHandler`. I don't think we should change the connectivity somehow manually in the unit test. This test would ensure we glue everything correctly together, because now we don't have an integration test.